indico/modules/categories/forms.py

```diff
@@ -171,3 +168,25 @@ def validate_entries(self, field):
                 raise ValidationError(_('Invalid category: {}').format(entry['id']))
             if entry['type'] == 'event' and not Event.get(entry['id'], is_deleted=False):
                 raise ValidationError(_('Invalid event: {}').format(entry['id']))
+
+
+class RoleForm(IndicoForm):
+    name = StringField(_('Name'), [DataRequired()],
+                       description=_('The full name of the role'))
+    code = StringField(_('Abbreviation'), [DataRequired(), Length(max=3)], filters=[lambda x: x.upper() if x else ''],
+                       render_kw={'style': 'width:60px; text-align:center; text-transform:uppercase;'},
+                       description=_('A shortcut (max. 3 characters) for the role'))
+    color = IndicoSinglePalettePickerField(_('Colour'), color_list=get_role_colors(), text_color='ffffff',
+                                           description=_('The colour used when displaying the role'))
+
+    def __init__(self, *args, **kwargs):
+        self.role = kwargs.get('obj')
+        self.category = kwargs.pop('category')
+        super(RoleForm, self).__init__(*args, **kwargs)
+
+    def validate_code(self, field):
+        query = CategoryRole.query.with_parent(self.category).filter_by(code=field.data)
+        if self.role is not None:
+            query = query.filter(CategoryRole.id != self.role.id)
+        if query.has_rows():
+            raise ValueError(_('A role with this abbreviation already exists.'))
```

Comment:
`A role with this code`?

Reply:
sounds better, i'd update this for event roles as well

Reply:
We are using `abbreviation`  in event role also and in both places, in UI it is named as `abbreviation`  not `code` .

Reply:
We use `code` in many other places for similar purposes.

Reply:
An apart from that we store it as `code` :P

Reply:
And we could just change the label for the `EventRole` field

Reply:
okay, I will make the change in both places. :+1: